app/redux/ducks/classify.js

```diff
@@ -315,7 +326,15 @@ export function fetchSubjects(workflow) {
         subjects,
         workflowID: workflow.id
       }
-    }));
+    }))
+    .catch((error) => {
+      if (error.type && error.type === 'Workflow complete') {
+        console.error(error);
+        dispatch(reset());
+      } else {
+        throw error
```

Comment:
Maybe re-throw the error in all cases. Then we can catch it in the parent component, and reset the stored workflow ID in user preferences too.